src/yunohost/domain.py

```diff
@@ -273,7 +273,7 @@ def domain_dns_conf(domain, ttl=None):
             '(?=.*(;[\s]*|")p=(?P<p>[^";]+))'), dkim_content, re.M|re.S
         )
         if dkim:
-            result += '\n{host}. {ttl} IN TXT "v={v}; k={k}; p={p}"'.format(
+            result += '\n{host} {ttl} IN TXT "v={v}; k={k}; p={p}"'.format(
```

Comment:
By removing the last dot you decide to use the "sub-domain notation", so `{host}` must not contain full domain, but only subdomain (`dkim.group('host')`).